pkg/querier/fanout/fanout.go

```diff
@@ -0,0 +1,127 @@
+package fanout
+
+import (
+	"context"
+
+	"github.com/grafana/loki/pkg/iter"
+	"github.com/grafana/loki/pkg/loghttp"
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/logql"
+	"github.com/grafana/loki/pkg/querier"
+	"github.com/grafana/loki/pkg/storage/stores/index/stats"
+)
+
+func NewQuerier(primary querier.Querier, secondaries ...querier.Querier) querier.Querier {
+	return &Querier{Querier: primary, secondaries: secondaries}
+}
+
+type Querier struct {
+	querier.Querier
+	secondaries []querier.Querier
+}
+
+func (q *Querier) SelectLogs(ctx context.Context, params logql.SelectLogParams) (iter.EntryIterator, error) {
+	if len(q.secondaries) == 0 {
+		return q.Querier.SelectLogs(ctx, params)
+	}
+	iters := make([]iter.EntryIterator, 0)
+	localIter, err := q.Querier.SelectLogs(ctx, params)
+	if err != nil {
+		return nil, err
+	}
+	iters = append(iters, localIter)
+	for _, remoteStore := range q.secondaries {
```

Comment:
We probably want to do these concurrently. Check out `github.com/grafana/dskit/concurrency` and the `ForEachJob` function specifically.

Reply:
done.thanks.